### PR TITLE
Added (emu-only) RCVI hack.

### LINKED ISSUE #359: RCVI Hack
Because some emulator hacks desperately need it

### PR DESCRIPTION
**Pull request by Reonu**
This should always be disabled by default.

Fixes #359

### CODE REVIEW

File: include/config/config_rom.h [hunk 1]
```diff
@@ -23,3 +23,7 @@
 // There is generally no reason to have a value other than 0 for emulator. As for console, it provides a (small) performance boost.
 #define BORDER_HEIGHT_CONSOLE  0
 #define BORDER_HEIGHT_EMULATOR 0
+
+// RCVI hack. Increases performance on emulator, and does nothing on console
+// Might break on some emulators. Use at your own risk, and don't use it unless you actually need the extra performance.
+// #define RCVI_HACK
```
[the remark this answers is not in the document]
Comment (Reonu) [reply include/config/config_rom.h:29]: > No newline moment

What do you mean?

[the remark this answers is not in the document]
Comment (gheskett) [reply include/config/config_rom.h]: Screenshot_20220627-142020_Chrome
That.

File: src/boot/main.c [hunk 2]
```diff
@@ -328,7 +328,7 @@ void thread3_main(UNUSED void *arg) {
         gBorderHeight = BORDER_HEIGHT_EMULATOR;
         gIsVC = IS_VC();
 #ifdef RCVI_HACK
-        VI.comRegs.vSync = 525*4;   
+        VI.comRegs.vSync = 525*20;
```
Comment (gheskett) [inline src/boot/main.c:331]: Idk much about how this works but is there any reason whatsoever you wouldn't want to use `525` as a constant here?

Comment (Reonu) [reply src/boot/main.c:331]: pretty sure it will compile to the exact same output

Comment (gheskett) [reply src/boot/main.c:331]: My thinking was different versions/regions but idk I guess this is emu only anyway (although it looks like it impacts VC, is that acceptable?)

Comment (Reonu) [reply src/boot/main.c:331]: yes it works on vc too

Comment (gheskett) [reply src/boot/main.c:331]: Alright I'll approve then

**Review by gheskett — APPROVED** [review]

**Review by gheskett — CHANGES_REQUESTED** [review]
image

Maybe should've compiled it _before_ I approved it...

**Review by gheskett — APPROVED** [review]
As long as moving the `change_vi` function doesn't kill us in the performance lottery for some reason, I have verified this works.

File: src/boot/main.c [hunk 3]
```diff
@@ -128,6 +128,24 @@ void create_thread(OSThread *thread, OSId id, void (*entry)(void *), void *arg,
 extern void func_sh_802f69cc(void);
 #endif
 
+void change_vi(OSViMode *mode, int width, int height){
```
Comment (arthurtilly) [inline src/boot/main.c:131]: uh any reason to move this function?

File: src/boot/main.c [hunk 4]
```diff
@@ -298,8 +316,11 @@ void handle_dp_complete(void) {
     sCurrentDisplaySPTask->state = SPTASK_STATE_FINISHED_DP;
     sCurrentDisplaySPTask = NULL;
 }
-extern void crash_screen_init(void);
 
+extern void crash_screen_init(void);
+#ifdef RCVI_HACK
+extern OSViMode VI;
```
Comment (arthurtilly) [inline src/boot/main.c:322]: the extern doesn't need to be ifdefed, externs have no effect on compilation if unused

**Review by arthurtilly — CHANGES_REQUESTED** [review]

File: src/boot/main.c (hunk 3 above)
Comment (gheskett) [reply src/boot/main.c:131]: iirc this was to avoid redefining it at the top of the function since it's not in a header I don't think

Comment (arthurtilly) [reply src/boot/main.c:131]: wouldnt it be easier and less conflicting to just add it in the header?

Comment (gheskett) [reply src/boot/main.c:131]: I generally don't like externing functions whenever it's not necessary but I suppose there isn't too much harm done here if we do.

**Review by gheskett — CHANGES_REQUESTED** [review]
Turns out the new RCVI value breaks the FPS counter (displays a value of 1.5, so probably just a matter of multiplying it by 20). Too lazy to actually fix this myself right now so I'm gonna request changes for the time being :)

**Comment by Reonu** [comment]
are we actually doing this, or are we scared about more accurate emulators and scrapping it?

**Comment by arthurtilly** [comment]
here we were waiting for arctic to fix the fps counter which he has said he will do multiple times and has not :pensive:

**Comment by Reonu** [comment]
i mean if the solution is as simple as he said I can literally just do it myself

**Comment by arthurtilly** [comment]
im pretty sure there were more complications regarding different emus giving different values, ask him about it

**Comment by gheskett** [comment]
Realistically I am pretty sure at this point that either we need to add a special custom detection system to correct it or we need to just merge as-is. The biggest issue I have isn't the FPS counter itself, but rather the OS time functions being way off.

I have to assume detection is possible by calling a start and end time function and comparing the differences, but I would need to play around with the best way to approach it in which it will always get a correct result, and in a way where it will always get called regardless of title screen usage and what not.

**Review by gheskett — APPROVED** [review]
Ready to be merged. I don't know how to fix the emulator profiler thing but at this point I think it's hopeless anyway.

**Pull request merged**